src/main/java/net/pms/dlna/DLNAMediaDatabase.java

```diff
@@ -196,22 +196,36 @@ public synchronized void init(boolean force) {
 							SwingUtilities.getWindowAncestor((Component) PMS.get().getFrame()),
 							String.format(Messages.getString("DLNAMediaDatabase.5"), dbDir),
 							Messages.getString("Dialog.Error"),
-							JOptionPane.ERROR_MESSAGE);
+							JOptionPane.ERROR_MESSAGE
+						);
 					}
 					LOGGER.error("Damaged cache can't be deleted. Stop the program and delete the folder \"" + dbDir + "\" manually");
 					PMS.get().getRootFolder(null).stopScan();
 					configuration.setUseCache(false);
 					return;
 				}
 			} else {
-				LOGGER.error("Database connection error: " + se.getMessage());
-				LOGGER.trace("", se);
-				RootFolder rootFolder = PMS.get().getRootFolder(null);
-				if (rootFolder != null) {
-					rootFolder.stopScan();
+				LOGGER.debug("Database connection error, retrying in 10 seconds");
+
+				try {
+					Thread.sleep(10000);
+					conn = getConnection();
+				} catch (InterruptedException | SQLException se2) {
+					if (!net.pms.PMS.isHeadless()) {
+						JOptionPane.showMessageDialog(
+							SwingUtilities.getWindowAncestor((Component) PMS.get().getFrame()),
```

Comment:
@SubJunk this causes

     java.lang.NullPointerException
	at javax.swing.SwingUtilities.getWindowAncestor(SwingUtilities.java:137)
	at net.pms.dlna.DLNAMediaDatabase.init(DLNAMediaDatabase.java:216)
	at net.pms.PMS.getDatabase(PMS.java:270)
	at net.pms.database.Tables.<clinit>(Tables.java:45)

Tested by using H2 console and start UMS.

Reply:
Ah yes that makes sense because I think the GUI isn't fully initialised at that point - it's just the splash page. Will fix

Reply:
That should be fixed now. I have left that dialog in there because that code is also called from the GUI as well as on startup

Reply:
BTW I am approving this because GitHub won't let you approve your own PR. Please merge if you approve